### DISCUSSION TITLE
Returning data from C++ on invokeMethod

### DISCUSSION THREAD

**Discussion by emanuele-f** [CefView/QCefView#190] [Ideas]
invokeMethod makes it convenient to call C++ code from Javascript. However it seems like it's not possible to return data to javascript. It would be great if invokeMethod allowed returning a string to Javascript.

Edit: I missed the `cefQueryRequest` method, which make it possible to accomplish this. I think it would be better to merge the methods into one, as getting a result from a call is a recurrent task

**Comment by tishion** [comment]
Hi @emanuele-f, glad to see you have solved the problem with **cefQueryRequest**

 > I think it would be better to merge the methods into one, as getting a result from a call is a recurrent task

unfortunately, this is not possible,  it's is up to the architecture of CEF, when you use invokeMethod from JS to call the method in C++, you must know the JS is executed in CEF **renderer** process while the C++ method is executed in **browser** process and CEF only provides an asynchronous IPC mechanism for communication.

**Reply by emanuele-f** [reply]
Hi, thanks for your reply. I've posted this only by reasoning from an API point of view, without prior knowledge of CEF. Now that I'm worked with CEF for a while, this makes sense to me. Closing this.
